Note that /logout doesn't take a body (#1644)
matrix-org/matrix-spec: content/client-server-api/_index.md

Clients are authenticated using opaque `access_token` strings (see [Client
Authentication](#client-authentication) for details).

All `POST` and `PUT` endpoints, with the exception of those listed below,
require the client to supply a request body containing a (potentially empty)
JSON object.  Clients should supply a `Content-Type` header of
`application/json` for all requests with JSON bodies, but this is not required.

The exceptions are:

> - [`POST /_matrix/media/v3/upload`](#post_matrixmediav3upload),
>   which takes any type of request body depending on what type of media is to be uploaded.
> - [`PUT /_matrix/media/v3/upload/{serverName}/{mediaId}`](#put_matrixmediav3uploadservernamemediaid),
>   which takes any type of request body depending on what type of media is to be uploaded.

Reviewer: I feel like this is getting a bit verbose. (And sorry, I know I previously encouraged you to spell this out more.)

Can we combine them into a single entry:

```suggestion
- [`POST /_matrix/media/v3/upload`](#post_matrixmediav3upload) and 
  [`PUT /_matrix/media/v3/upload/{serverName}/{mediaId}`](#put_matrixmediav3uploadservernamemediaid),
  both of which take the uploaded media as the request body.
```

> - [`POST /_matrix/client/v3/logout`](#post_matrixclientv3logout),
>   which does not take any request body.

Reviewer: logout/all has the same problem:

```suggestion
- [`POST /_matrix/client/v3/logout`](#post_matrixclientv3logout) and
  [`POST /_matrix/client/v3/logout/all`](#post_matrixclientv3logoutall),
  which take an empty request body.
```

Authentication](#client-authentication) for details).

All `POST` and `PUT` endpoints, with the exception of [`POST
/_matrix/media/v3/upload`](#post_matrixmediav3upload), [`PUT
/_matrix/media/v3/upload/{serverName}/{mediaId}`](#put_matrixmediav3uploadservernamemediaid)

> and [`POST /_matrix/client/v3/logout`](#post_matrixclientv3logout),

Reviewer: it might be worth emphasising what the exceptions do require.
